hometask_11.py

```diff
@@ -0,0 +1,82 @@
+# Помилки та винятки
+# part 1
+# Написати власний декоратор, задачею якого має бути друк назви функції і часу, коли вона була викликана.
+# Декоратор має працювати для різних функцій однаково.
+
+
+from datetime import datetime
+
+def own_dec(func):
+    def time_func(*args, **kwargs):
+        print(f"'{func.name}' find that now is {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
+        func(*args, **kwargs)
```

Comment:
use `return` statement here to be able to decorate functions that return some value

Reply:
from datetime import datetime

def own_dec(func):
    def time_func(*args, **kwargs):
        print(f"'{func.__name__}' find that now is {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
        func(*args, **kwargs)
    return time_func

@own_dec
def func_1(a, b):
    print(f"func_1 find sum of a and b = {a + b}")

@own_dec
def func_2(c, d):
    print(f"func_2 find multiplication of c * d = {c * d}")

func_1(2, 2)
func_2(2, 4)

Reply:
here `return func(*args, **kwargs)`, line 12

Reply:
from datetime import datetime

def own_dec(func):
def time_func(*args, **kwargs):
print(f"'{func.name}' find that now is {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
        result = func(*args, **kwargs)
        return result
return time_func

Reply:
Pls check my updates